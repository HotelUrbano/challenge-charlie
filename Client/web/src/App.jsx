import { memo, useCallback, useEffect, useState } from "react";

import "./Styles/global.css";
import { AppStyle } from "./Styles/app";
import env from "react-dotenv";
import GoogleService from "./Services/google";
import { CardWeather } from "./Components/Card";

export const App = memo(() => {
    const [nameLocale, setNameLocale] = useState(null);
    useEffect(() => {
        async function getLocation() {
            if (!("geolocation" in navigator)) {
                return alert("Navegador nÃ£o compativel com geolocalizaÃ§Ã£o");
            }

            navigator.geolocation.getCurrentPosition(
                async ({ coords: { latitude, longitude } }) => {
                    const {
                        data: { results },
                    } = await GoogleService.getLocationGoogle(
                        latitude,
                        longitude,
                        env.KEY_API
                    );

                    setNameLocale(
                        `${results[6].address_components[0].long_name}, ${results[6].address_components[1].long_name}`
                    );

                    console.log("results", results);
                },
                (error) => {
                    return alert(`Error: ${error}`);
                }
            );
        }

        getLocation();
    }, []);

    const handleChange = useCallback((value) => {}, []);

    const handleOnBlur = useCallback(() => {}, []);
    return (
        <>
            <AppStyle>
                <CardWeather
                    nameLocale={nameLocale}
                    handleChange={handleChange}
                    handleOnBlur={handleOnBlur}
                />
            </AppStyle>
        </>
    );
});
