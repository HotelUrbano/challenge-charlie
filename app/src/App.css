.App .wrapper {
  background-position: center;
  background-size: cover; }

.App .container {
  margin: 0 auto;
  display: block;
  max-width: 600px;
  width: 100%;
  border-radius: 10px;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center; }

.App .loader {
  background-color: #eeeeee85;
  border-radius: 0;
  margin: 0 auto;
  display: block;
  max-width: 600px;
  width: 100%;
  min-height: 80vh;
  display: flex;
  flex-direction: column;
  justify-content: center; }

.App .weather {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  margin-bottom: 10px; }

.App .forecast {
  text-align: center; }
