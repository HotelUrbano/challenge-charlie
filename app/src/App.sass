.App
    .wrapper
        background-position: center
        background-size: cover

    .container
        margin: 0 auto
        display: block
        max-width: 600px
        width: 100%
        border-radius: 10px
        min-height: 100vh
        display: flex
        flex-direction: column
        justify-content: center
    .loader
        background-color: #eeeeee85
        border-radius: 0
        margin: 0 auto
        display: block
        max-width: 600px
        width: 100%
        min-height: 80vh
        display: flex
        flex-direction: column
        justify-content: center
    .weather
        display: flex
        flex-wrap: wrap
        justify-content: space-around
        align-items: center
        margin-bottom: 10px
    .forecast
        text-align: center
