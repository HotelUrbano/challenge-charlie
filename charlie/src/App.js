import React from "react";
import Weather from "../src/components/Weather/Weather";
import Background from '../src/components/Background/Background';


function App() {
  return (
    <div className="App">
        <Background />
          
      <Weather />
    </div>
  );
}

export default App;