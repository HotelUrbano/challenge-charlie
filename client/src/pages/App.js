import React, { Component } from 'react';
import { connect } from 'react-redux';
// Action(s)
import * as actions from '../actions';
import SearchSvg from '../assets/images/search.svg';
// Component
import WeatherCity from '../components/Weather';
import Spinner from '../components/Spinner';

// Para temperaturas abaixo de 15ºC deve ser usado tons de azul,
// para temperaturas acima de 35ºC deve ser usado tons de vermelho
// e use tons de amarelo para as demais temperaturas
class App extends Component {
    state = { inputFocused: false, city: '', convertedTemp: true };

    componentDidMount() {
        this.askForPosition().then(({ latitude, longitude }) => this.props.fetchLatLongWeather(latitude, longitude));
        this.props.fetchBingData();
        this.props.fetchCityWeather();
    }

    askForPosition = () => {
        return new Promise((resolve, reject) => {
            navigator.geolocation.getCurrentPosition(({ coords }) => resolve(coords), error => reject(error));
        });
    };

    handleSubmit = e => {
        e.preventDefault();
        this.props.fetchCityWeather(this.state.city);
    };

    handleConvertTemp = () => {
        this.setState({ convertedTemp: !this.state.convertedTemp });
    };

    render() {
        const { city, loading, bkgImg } = this.props;
        const { convertedTemp } = this.state;

        return (
            <main
                className="main"
                style={{
                    backgroundImage: `url(${bkgImg.image})`
                }}
            >
                <div className={`backdrop ${this.state.inputFocused ? 'backdrop--active' : ''}`} />
                <section className="container home">
                    <h1 className="home__name">Charlie Weather</h1>
                    <form className="search" onSubmit={this.handleSubmit}>
                        <input
                            type="text"
                            className="search__input"
                            onChange={e => this.setState({ city: e.target.value })}
                            placeholder="Search for a city"
                            onFocus={() => this.setState({ inputFocused: true })}
                            onBlur={() => this.setState({ inputFocused: false })}
                        />
                        <button className="search__button" type="submit">
                            <img src={SearchSvg} alt="ButtonSearch" />
                        </button>
                    </form>
                    {loading ? (
                        <Spinner />
                    ) : Object.keys(city).length <= 1 ? (
                        <h1 className="heading-1 error-text">Oops!</h1>
                    ) : (
                        <div className="weather-box">
                            <div className="weather-box__header">
                                <h2 className="heading-2">{city.location.city}</h2>
                                <div className="weather-box__header__group">
                                    <h4 className="heading-4">{city.location.country}</h4>
                                    <h4 className="heading-4">{city.lastBuildDate}</h4>
                                </div>
                            </div>
                            <div className="animate-top">
                                {city.item.forecast.map((day, i) => {
                                    if (i < 3) {
                                        return (
                                            <WeatherCity
                                                key={i}
                                                convertTemp={this.handleConvertTemp}
                                                cityWeather={{ day, city }}
                                                index={i}
                                                tempFormatted={convertedTemp}
                                            />
                                        );
                                    }
                                })}
                            </div>
                        </div>
                    )}
                </section>
            </main>
        );
    }
}

const stateToProps = ({ weather }) => weather;

export default connect(
    stateToProps,
    actions
)(App);
