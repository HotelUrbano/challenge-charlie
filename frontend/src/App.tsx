import React from 'react';

import Layout from '@components/Layout';
import Home from '@pages/Home';

import '@theme/main.scss';

const App: React.FC = () => (
  <Layout>
    <Home />
  </Layout>
);

export default App;

/**
 * @see https://gist.github.com/hcatlin/1027867
 */
