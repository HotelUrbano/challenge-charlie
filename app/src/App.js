import React, { Component } from 'react';
import './App.css';
import axios from 'axios';
import _ from 'lodash';
import SearchBar from './components/SearchBar';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            bgBing: '',
            location: '',
            results: '',
            forecast: [],
            error: ''
        };
    }

    componentDidMount() {
        axios.get('https://bingwallpaper.herokuapp.com').then(res => {
            let bgImage = res.data;
            console.log(res);
            this.setState({
                bgBing: bgImage
            });
        });
        this.getGeoLocation();
    }

    getGeoLocation = () => {
        const findLocation = position => {
            let coordinates =
                position.coords.latitude + ',' + position.coords.longitude;
            let queryLocation = `https://query.yahooapis.com/v1/public/yql?q=select * from weather.forecast where woeid in (SELECT woeid FROM geo.places WHERE text="(${coordinates})")&format=json`;
            axios.get(queryLocation).then(resp => {
                this.setState({
                    location: resp.data.query.results.channel.location.city,
                    forecast: resp.data.query.results.channel.item.forecast
                });
            });
        };
        const errorHandler = error => {
            this.setState({
                error: error
            });
        };

        navigator.geolocation.getCurrentPosition(findLocation, errorHandler);
    };

    inputHandler = e => {
        this.setState({ location: e.target.value });
        this.getByInputLocation();
    };

    getByInputLocation = _.debounce(
        () => {
            let queryLocation = `https://query.yahooapis.com/v1/public/yql?q=select * from weather.forecast where woeid in (SELECT woeid FROM geo.places WHERE text="(${
                this.state.location
            }, BR)")&format=json`;
            axios.get(queryLocation).then(resp => {
                this.setState({
                    results: resp.data.query.results.channel
                });
            });
        },
        1000,
        { maxwait: 1000 }
    );

    render() {
        let sytleBg = {
            backgroundImage: `url(${this.state.bgBing})`,
            width: '100%',
            height: '100vh',
            backgroundPosition: 'center',
            backgroundSize: 'cover'
        };
        return (
            <div className="App" style={sytleBg}>
                <div className="container">
                    <SearchBar
                        location={this.state.location}
                        change={this.inputHandler}
                    />
                </div>
            </div>
        );
    }
}

export default App;
