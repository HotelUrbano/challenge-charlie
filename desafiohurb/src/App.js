import React, { useState, useEffect, useCallback } from "react";
import './App.css';
import { toFahrenheit, toCelsius } from './utils/utils';
import { OpenWeatherService } from './services/openweathermap';
import { OpenCageService } from './services/opencagedata';

function App() {
    const [textInput, setTextInput] = useState('');
    const [coords, setCoords] = useState({
        latitude: null,
        longitude: null
    });
    const [tomorow, setTomorow] = useState();
    const [afterTomorow, setAfterTomorow] = useState();
    const [backgroundImage, setBackgroundImage] = useState();
    const [today, setToday] = useState({});
    const [isCelsius, setIsCelsius] = useState(true);
    const [error, setError] = useState(false);
    const [loading, setLoading] = useState(true);



    useEffect(() => {
        if (!coords.latitude || !coords.longitude) {
            return;
        }
        OpenCageService.getCityByLongitude(coords).then(response => {
            setTextInput(response);
        });
    }, [coords]);

    function changeTemperature() {
        if (isCelsius) {
            setToday({
                ...today,
                temp: toFahrenheit(today.temp).toFixed(0),
            });
            setTomorow(toFahrenheit(tomorow).toFixed(0));
            setAfterTomorow(toFahrenheit(afterTomorow).toFixed(0));
        } else {
            setToday({
                ...today,
                temp: toCelsius(today.temp).toFixed(0),
            });
            setTomorow(toCelsius(tomorow).toFixed(0));
            setAfterTomorow(toCelsius(afterTomorow).toFixed(0));
        }
        setIsCelsius(!isCelsius);
    }

    function handleChange(e) {
        var text = e.target.value;
        setTextInput(text);
    }
    return (
        <div className="App" style={{ backgroundImage: `url(${backgroundImage})` }}>
            <div className="container">


                <section className="search">
                    <input data-icon="(" placeholder="Digite uma cidade..." className="search__input" type="text" value={textInput} onChange={handleChange} />
                </section>
                {error ? (<section className={`error background-none`}><h1>{loading ? "Carregando..." : "Pesquise outra cidade!"}</h1></section>) : <>
                    <section className={`today `}>
                        <div className="today__box">
                            <div className="today__icon">
                                <span>{today.icon}</span>
                            </div>
                            <div className="today__box--data">
                                <div className="temperature">
                                    <span>HOJE</span>
                                    <span className="temperature--action" onClick={changeTemperature} >{today.temp}{isCelsius ? "ºC" : "ºF"}</span>
                                </div>
                                <div className="climate">
                                    <span className="climate__description">{today.description}</span>
                                    <span>Vento:NO {today.wind}km/h</span>
                                    <span>Humidade: {today.humidity}%</span>
                                    <span>Pressão: {today.pressure}hPA</span>
                                </div>
                            </div>
                        </div>
                    </section>

                    <section className={`tomorow`}>
                        <div className="tomorow__text">
                            <span>AMANHÃ</span>
                            <span className="temperature--action" onClick={changeTemperature} >{tomorow}{isCelsius ? "ºC" : "ºF"}</span>
                        </div>
                    </section>

                    <section className={`after-tomorow`}>
                        <div className="after__text">
                            <span>DEPOIS DE AMANHÃ</span>
                            <span className="temperature--action" onClick={changeTemperature}>{afterTomorow}{isCelsius ? "ºC" : "ºF"}</span>
                        </div>
                    </section>
                </>}


            </div>
        </div>
    );
}

export default App;
