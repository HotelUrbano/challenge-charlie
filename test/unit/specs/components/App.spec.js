import { mount } from '@vue/test-utils'
import sinon from 'sinon'
import axios from 'axios'
import flushPromises from 'flush-promises'
import App from '@/App.vue'
import MockApiBing from '../../../mocks/bing.json'
import { BING_URL } from '@/services/BingService'

describe('App.vue', () => {
  let instanciaComponente

  const STUB = sinon.stub(axios, 'get')
    .returns(Promise.resolve({
      data: MockApiBing
    }))

  beforeEach(async () => {
    instanciaComponente = mount(App, {
      stubs: ['router-view']
    })
    await flushPromises()
  })

  afterEach(() => {
    instanciaComponente.destroy()
    STUB.resetHistory()
  })

  it('Deve renderizar sem falhas', () => {
    expect(instanciaComponente.exists()).toBe(true)
    expect(instanciaComponente.html()).toMatchSnapshot()
  })

  it('Deve renderizar o background corretamente', () => {
    instanciaComponente.vm.$nextTick(() => {
      expect(STUB.calledOnce).toBe(true)
      expect(instanciaComponente.vm.urlImagemBackground).toEqual(`${BING_URL}${MockApiBing.images[0].url}`)
      expect(instanciaComponente.vm.style).toBe(`background-image: url(${instanciaComponente.vm.urlImagemBackground})`)
    })
  })
})
