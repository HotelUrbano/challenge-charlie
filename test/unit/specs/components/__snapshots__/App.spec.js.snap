// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Deve renderizar sem falhas 1`] = `
<section id="app" class="app" style="background-image: url(https://bing.com/th?id=OHR.DenaliCaribou_ROW5992360682_1920x1080.jpg&amp;rf=LaDigue_1920x1080.jpg&amp;pid=hp);">
    <router-view-stub></router-view-stub>
</section>
`;
